### DISCUSSION TITLE
Horizontal layout for US NSPanel?

### DISCUSSION THREAD

**Discussion by alexruffell** [Blackymas/NSPanel_HA_Blueprint#386] [General]
This is an awesome project! I had set aside the NSPanel as it was hard to get everything to work especially being unfamiliar with the software that makes the HMI / tft files. This project makes it so much easier!

I have a vertical US style NSPanel that I'd like to mount sideways and use it as a Media Room light controller. A sideways layout would work much better if sitting at a 45 degree angle inside a custom made holder on a table next to the sofa. I realize I can modify the HMI file but I was hoping something was already there and I am also concerned with then having to figure out how to keep it in line (feature parity) with any future updates released as part of this project.

**Comment by abidullah** [comment]
pm me i can send u my tft file

**Reply by Blackymas** [reply]
it would also be interesting for me what you have changed,

then I could try to take it over

**Reply by alexruffell** [reply]
@Blackymas I believe @abidullah kindly shared his modified version here: https://github.com/abidullah/NSPanel_HA_Blueprint

The readme doesn't specifically say horizontal layout but he pointed me to that. I have yet to get a chance to try it out. The modified HMI and TFT files are in a sub folder. 

Either way, thanks to both for sharing your work!

**Reply by Blackymas** [reply]
Interesting - but if I know exactly what needs to be changed I can provide a version.

The reason is with version 3.2 has also changed on the part of HMI some. Either @abidullah changes it then again or explains to me what did not fit ;)

**Reply by alexruffell** [reply]
> what did not fit ;)

I am guessing that he had to flip all the elements sideways and move stuff around so a completely new HMI in terms of layout... unless he was able to reuse the EU layout. The EU screen is slightly shorter so that would leave about 0.5cm unused on the bottom if I recall correctly.

**Comment by abidullah** [comment]
one more thing to point at the nspanel_esphome.yaml 
there is one small mistake i hope this too can create a problem

in ur code at the line 18 ur code is
`  nextion_update_url: !secret nspanel_update_url # add in your esphome secrets file. Example: "http://"HOME ASSISTANT IP":8123/local/nspanel/nspanel.tft"`

![image](https://user-images.githubusercontent.com/2981996/223547809-8119c150-3e85-4b1c-b4ed-ee7ff8210e47.png)

and the doubt is the variable   **Example:**  with Colon ( : ) where the Esphome compiler thinks it as a variable
which should be  **Example** without Colon ( : )

`  nextion_update_url: !secret nspanel_update_url # add in your esphome secrets file. Example "http://"HOME ASSISTANT IP":8123/local/nspanel/nspanel.tft"`

![image](https://user-images.githubusercontent.com/2981996/223547933-55de258c-6cc1-4495-bfea-8b414cd1741d.png)

**Reply by Blackymas** [reply]
not sure what you mean

**Reply by abidullah** [reply]
see my screenshot attached 

ur code has Colon ( : ) at line 18 column 95 Example: "http://"HOME ASSISTANT IP":8123/local/nspanel/nspanel.tft"
which should be Example "http://"HOME ASSISTANT IP":8123/local/nspanel/nspanel.tft" 
i guess

**Reply by Blackymas** [reply]
ah - but you use a very old version ;)
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-First-Steps---Installation-and-Setup

**Comment by abidullah** [comment]
YesRegards,Aftab Mohammed.On 8 Mar 2023, at 3:28 AM, Blackymas ***@***.***> wrote:﻿
ah - but you use a very old version ;)
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-First-Steps---Installation-and-Setup

**Comment by abidullah** [comment]


**Reply by Blackymas** [reply]
This is a very good idea.
I've already had this request twice but never had the time to implement it.

If you find a solution I will be happy to implement it.

**Reply by abidullah** [reply]
@Blackymas  code is completed with the Option 12hr and 24hr format.
please check 
attached and updated [nspanel_blueprint.yaml](https://github.com/Abidullah/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml)
and uploaded the HMI and TFT file too

**Reply by alexruffell** [reply]
@Blackymas Will you by any chance be implementing @abidullah 's horizontal layout solution?

**Reply by robsonfelix** [reply]
that would be really nice... I have a handful of NSPanels I was using horizontally with eWeLink that I need migrated to Home Assistant. I have already migrated all others... roughly 15 of them.

**Comment by edwardtfn** [comment]
Version 4.0 will include a TFT for US model in landscape mode. And a lot of other awesome things.
It is already in `dev` and will be released soon.

**Comment by edwardtfn** [comment]
It's out for `beta` now: #1046

**Comment by edwardtfn** [comment]
I'm closing this discussion as we have released v4.0 (#1057).
Please add a feature request for the missing things.

And thanks a lot for your support.
